SwiftCompilerSources/Sources/Basic/Utils.swift

```diff
@@ -114,6 +114,37 @@ public struct StringRef : CustomStringConvertible, NoReflectionChildren {
   public static func ~=(pattern: StaticString, value: StringRef) -> Bool { value == pattern }
 }
 
+//===----------------------------------------------------------------------===//
+//                      Single-Element Inline Cache
+//===----------------------------------------------------------------------===//
+
+public struct SingleInlineCache<Element>: RandomAccessCollection {
```

Comment:
Right, it's an array not a cache. For now, it's for a very specific but important use case of a single inline element. Many of the APIs I need almost always have a single result, but sometimes more. It can be replaced by a more general `SmallArray` in the future, which might want to support contiguous memory. SmallArray would be an inappropriately vague name for this thing which has a very niche use case. It's not like C++ where we would use this everywhere, because this has limitations.

Reply:
How about `SingleInlineArray` or `SingleInlineCacheArray`?

Reply:
I change it to `SingleInlineArray` in the previous round.